### PR TITLE
implement full width dashboard properly

### PR DESCRIPTION
**Pull request by adrianjost**
Reimplements the full widht layout of the dashboard properly

- adds a new full-width layout
- adjust the lecture view to use new layout and wrap most elements into a container to center it again like it was before ([https://getbootstrap.com/docs/4.4/layout/overview/#containers](https://getbootstrap.com/docs/4.4/layout/overview/#containers))

image

### CODE REVIEW

**Comment by MartinBuessemeyer** [comment]
@adrianjost please resolve the merge conflict

**Comment by adrianjost** [comment]
@MartinBuessemeyer done

File: app/views/lectures/show.html.erb [hunk 1]
```diff
@@ -1,51 +1,54 @@
-<h5><%= @lecture.name %></h5>
-<% if @lecture.start_time &&  @lecture.end_time && @lecture.date%>
-  <p><%= @lecture.date.strftime("%d.%m.%Y") %> <%= @lecture.start_time.strftime("%H:%M -") %> <%= @lecture.end_time.strftime("%H:%M") %></p>
-<% end %>
+<div class="container">
```
Comment (MartinBuessemeyer) [inline app/views/lectures/show.html.erb]: The whole Dashboard is missing and 16 tests are failing. All failing tests are in the lectures/show.html.erb_spec.rb. I guess they should be fixed if you add the Dashboard

Comment (MartinBuessemeyer) [reply app/views/lectures/show.html.erb]: image

**Comment by MartinBuessemeyer** [comment]
I just updated my local branch to see if the tests are still failing and indeed they still exist.

File: app/views/layouts/application-full.html.erb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <title>LecturePortal</title>
+    <%= csrf_meta_tags %>
+    <%= csp_meta_tag %>
+    <!-- Bootstrap meta tags -->
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
```
[the remark this answers is not in the document]
Comment (adrianjost) [reply app/views/layouts/application-full.html.erb]: I just copied the default application.html.erb and changed line 21 from `container` to `container-fluid` + a container wrapper for the alerts.

I woulf prefer not to change anything else here to keep it consistent.

File: app/views/lectures/show.html.erb (hunk 1 above)
Comment (adrianjost) [reply app/views/lectures/show.html.erb]: sorry, that must have happend during the latest merge. Haven't tested it. I will fix this!

**Comment by MartinBuessemeyer** [comment]
Sorry for the criticism. I just want to mention that everything looks good :)
If you have time: the browse button for file upload is not bootstrapped yet. Can you maybe add this as well in this PR?

File: app/views/courses/show.html.erb [hunk 3]
```diff
@@ -47,10 +47,12 @@
...
-<% if !@current_user.is_student %>
-  <%= link_to 'Edit', edit_course_path(@course) %> |
-<% end %>
-<%= link_to 'Back', root_path %>
+<div class="d-flex justify-content-center">
+  <% if @current_user.is_student %>
+    <%= link_to "Leave", leave_course_path(id: @course.id), method: :post, class:"btn btn-primary" %> |
+  <% end %>
+  <% if !@current_user.is_student %>
+    <%= link_to 'Edit', edit_course_path(@course) %> |
```
[the remark this answers is not in the document]
Comment (adrianjost) [reply app/views/courses/show.html.erb]: done

File: app/views/layouts/application-full.html.erb (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrianjost) [reply app/views/layouts/application-full.html.erb]: moved all the head content into a partial

File: app/views/lectures/show.html.erb (hunk 1 above)
Comment (adrianjost) [reply app/views/lectures/show.html.erb]: Yeah the navbar rendering got lost during merge. Should be fixed now.

**Comment by adrianjost** [comment]
> Sorry for the criticism. I just want to mention that everything looks good :)
> If you have time: the browse button for file upload is not bootstrapped yet. Can you maybe add this as well in this PR?

I would prefer to do this in another small PR to keep it clean and easy to track what's changed.

**Comment by MartinBuessemeyer** [comment]
sure :)

File: app/views/courses/show.html.erb (hunk 3 above)
[the remark this answers is not in the document]
Comment (MartinBuessemeyer) [reply app/views/courses/show.html.erb]: sorry I know that I am annoying but please add some space between the links.

[the remark this answers is not in the document]
Comment (MartinBuessemeyer) [reply app/views/courses/show.html.erb]: or maybe swop the order: `<- Back | Edit`

[the remark this answers is not in the document]
Comment (MartinBuessemeyer) [reply app/views/courses/show.html.erb]: ok since the pr is merged by mergify lets make this another issue.

**Pull request merged**